editor/src/components/editor/actions/actions.tsx

```diff
@@ -1873,10 +1885,28 @@ export const UPDATE_FNS = {
             return path
           })
 
-        const withElementDeleted = deleteElements(staticSelectedElements, editor)
+        const { editor: withElementDeleted, affectedElements } = deleteElements(
+          staticSelectedElements,
+          editor,
+        )
         const parentsToSelect = uniqBy(
           mapDropNulls((view) => {
             const parentPath = EP.parentPath(view)
+            const oldParent = MetadataUtils.findElementByElementPath(editor.jsxMetadata, parentPath)
+            if (
+              oldParent != null &&
+              isRight(oldParent.element) &&
+              isJSXConditionalExpression(oldParent.element.value)
+            ) {
+              const isTrueBranch = EP.toUid(view) === getUtopiaID(oldParent.element.value.whenTrue)
+              const newParent = affectedElements[EP.toString(parentPath)]
```

Comment:
This is the original parent of the deleted element, isn't this in the original `editor.jsxMetadata`?

Reply:
ok, i get it!
But then we could also just store the original element path to the branch, am I right?

Reply:
yeah, for sure, it just seemed cleaner this way to me as when deleting you're always dealing with a parent element. normally deleting will remove "something" from a parent, the `null` case for conditionals is a special case. with the current setup you just have a list of the parent elements that were affected by the deletion, which you can then inspect as you wish (in this case, to get the new branch's ID)

Reply:
Moreover, we can just generated the new element path on the fly, can't we? We just have to append the uid of the new null expression to the original parent path, don't we?

Reply:
refactored to actually walk the project contents instead of returning the elements, it's definitely a better solution but I have no idea why it did not work before, I probably did the mistake of using the original project contents and never retry when I tested using the `withElementDeleted` state ¯\_(ツ)_/¯